cip/CIP2015-06-24-call-procedures.adoc

```diff
@@ -0,0 +1,250 @@
+RETURN n.name AS name
+----
+
+This will behave in the same way as above but since the procedure now returns nothing, calling it will not change the result cardinality, i.e. conceptually the procedure is called once for each record produced by `MATCH` and will just pass that record on to `RETURN`.
+
+==== Standalone calls
+
+Procedures may also be called standalone, i.e. without taking arguments from or combining their results with other parts of a larger query.
+In this case, the trailing `RETURN` clause may be omitted and all variable in scope are returned implicitly by the statement.
+Furthermore, we propose two optional simplifications that are available for standalone calls only.
+These simplifications are outlined in the following:
+
+===== Calling with implicitly passed arguments (parameters)
+
+Standalone calls may ommit passing arguments explicitly.  In this case, all required procedure arguments are taken implicitly from statement parameters with the same name.
+
+Again consider a call to the stored procedure `myProc(name``::``STRING?``,``id``::``INTEGER?)``::`` (last``::``STRING?)`:
+
+.Standalone call to a stored procedure using implicitly passed arguments
+[source, cypher]
+----
+CALL myProc YIELD last AS lastLogin
+----
+
+This is the same as executing:
+
+.Standalone call to a stored procedure using explicitly passed arguments
+[source, cypher]
+----
+CALL myProc({name}, {id}) YIELD last AS lastLogin
+----
+
+===== Calling without `YIELD`
+
+Standalone calls that ommit the `YIELD` subclause will always project out all result fields.
+
+Again consider a call to the stored procedure `myProc(name``::``STRING?``,``id``::``INTEGER?)``::`` (last``::``STRING?)`:
+
+.Standalone call to a stored procedure without `YIELD`
+[source, cypher]
+----
+CALL myProc("Donald", 12)
+----
+
+This is the same as executing:
+
+.Equivalent in-query call for standalone call to a stored procedure without `YIELD`
+[source, cypher]
+----
+CALL myProc("Donald", 12) YIELD last
+RETURN *
+----
+
+===== Calling with implicitly passed arguments (parameters) and without `YIELD`
+
+Both simplifications may be used in a single standalone procedure call, leading to a very concise syntax for just executing a single procedure call:
+
+.Simplified standalone procedure call
+[source, cypher]
+----
+CALL myProc
+----
+
+=== Procedure call semantics
+
+The technical details of how procedure calls are executed are left to the implementation as long as they do not violate
+the following rules:
+
+It is an error if calling a procedure fails to yield results in accordance with its declared result signature.
+
+A standalone call to a procedure that yields nothing returns no result records for the whole Cypher statement.
+
+An in-query call to a procedure that yields nothing executes the whole procedure without projecting any result fields and without changing the cardinality of the input record (i.e. all incoming records just pass through).
```

Comment:
"procedure that yields nothing"... may be a bit ambiguous at this juncture, in that this may be conflated with "void procedures". Can it be made clearer? Also, the same applies in the succeeding bullet points...